daras_ai_v2/base.py

```diff
@@ -1594,23 +1661,46 @@ def render_extra_waiting_output(self):
     def estimate_run_duration(self) -> int | None:
         pass
 
-    def submit_and_redirect(self):
+    def submit_and_redirect(self) -> typing.NoReturn | None:
         sr = self.on_submit()
         if not sr:
             return
         raise gui.RedirectException(self.app_url(run_id=sr.run_id, uid=sr.uid))
 
+    def publish_and_redirect(self) -> typing.NoReturn | None:
+        assert self.request.user and not self.request.user.is_anonymous
+
+        updated_count = SavedRun.objects.filter(
+            id=self.current_sr.id,
+            uid=self.current_sr.uid,
+            # filter for RecipeRunState.standby
+            run_status="",
+            error_msg="",
+            run_time=datetime.timedelta(),
+        ).update(run_status=STARTING_STATE, uid=self.request.user.uid)
+        if updated_count >= 1:
+            # updated now
+            self.dump_state_to_sr(self._get_validated_state(), self.current_sr)
```

Comment:
not sure why this is needed here?

Reply:
ah, good catch. i think it is leftover from the earlier hack where the state was carried over without creating a saved run. thanks!